posthog/warehouse/data_load/service.py

```diff
@@ -89,8 +89,13 @@ def to_temporal_schedule(
     )
 
     # Determine spec based on frequency
-    if sync_frequency <= timedelta(hours=1):
-        spec = ScheduleSpec(intervals=[ScheduleIntervalSpec(every=sync_frequency)], jitter=jitter)
+    if sync_frequency <= timedelta(hours=1) or sync_frequency > timedelta(days=1):
+        spec = ScheduleSpec(
+            intervals=[
+                ScheduleIntervalSpec(every=sync_frequency, offset=timedelta(hours=hour_of_day, minutes=minute_of_hour))
+            ],
+            jitter=jitter,
+        )
```

Comment:
Hard to know what this is actually doing tbh, could we get some tests here?

Reply:
I had the same thought, so am working on a PR to add some tests to test the schedules in Temporal actually look as we'd expect, by inspecting the upcoming scheduled runs